Panic when using amtool

```
$ ./amtool --alertmanager.url=http://127.0.0.1:9093 alert add foo=
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/prometheus/alertmanager/pkg/labels.ParseMatcher({0x7fffcacb016f, 0x4})
        /home/roidelapluie/dev/alertmanager/pkg/labels/parse.go:130 +0x67f
github.com/prometheus/alertmanager/cli.parseLabels({0xc000569c38, 0x1, 0xc00007c570?})
        /home/roidelapluie/dev/alertmanager/cli/utils.go:110 +0x7b
github.com/prometheus/alertmanager/cli.(*alertAddCmd).addAlert(0xc00005d080, {0x103c0b0?, 0xc00080c0c0}, 0x6fc23ac00?)
        /home/roidelapluie/dev/alertmanager/cli/alert_add.go:76 +0x97
github.com/prometheus/alertmanager/cli.execWithTimeout.func1(0x40b4d4?)
        /home/roidelapluie/dev/alertmanager/cli/utils.go:154 +0x70
gopkg.in/alecthomas/kingpin%2ev2.(*actionMixin).applyActions(0xc000286150?, 0xcfd2c0?)
        /home/roidelapluie/go/pkg/mod/gopkg.in/alecthomas/kingpin.v2@v2.2.6/actions.go:28 +0x5c
gopkg.in/alecthomas/kingpin%2ev2.(*Application).applyActions(0xc0000de870?, 0xc0002ce090)
        /home/roidelapluie/go/pkg/mod/gopkg.in/alecthomas/kingpin.v2@v2.2.6/app.go:557 +0xed
gopkg.in/alecthomas/kingpin%2ev2.(*Application).execute(0xc0000de870, 0xc0002ce090?, {0xc000598040, 0x2, 0x2})
        /home/roidelapluie/go/pkg/mod/gopkg.in/alecthomas/kingpin.v2@v2.2.6/app.go:390 +0x66
gopkg.in/alecthomas/kingpin%2ev2.(*Application).Parse(0xc0000de870, {0xc0000300b0?, 0xc0000300b0?, 0x4?})
        /home/roidelapluie/go/pkg/mod/gopkg.in/alecthomas/kingpin.v2@v2.2.6/app.go:222 +0x14f
github.com/prometheus/alertmanager/cli.Execute()
        /home/roidelapluie/dev/alertmanager/cli/root.go:162 +0xd45
main.main()
        /home/roidelapluie/dev/alertmanager/cmd/amtool/main.go:19 +0x17

```

When building `main`

--- a/pkg/labels/parse.go
+++ b/pkg/labels/parse.go
@@ -120,8 +120,12 @@ func ParseMatcher(s string) (_ *Matcher, err error) {
 		return nil, errors.Errorf("bad matcher format: %s", s)
 	}
 
+	rawValue := ms[3]
+	if len(rawValue) == 0 {
+		return nil, errors.Errorf("matcher value is not present: %s", s)
+	}
+
 	var (
-		rawValue            = ms[3]
 		value               strings.Builder
 		escaped             bool
 		expectTrailingQuote bool

--- a/pkg/labels/parse.go
+++ b/pkg/labels/parse.go
@@ -120,18 +120,14 @@ func ParseMatcher(s string) (_ *Matcher, err error) {
 		return nil, errors.Errorf("bad matcher format: %s", s)
 	}
 
-	rawValue := ms[3]
-	if len(rawValue) == 0 {
-		return nil, errors.Errorf("matcher value is not present: %s", s)
-	}
-
 	var (
+		rawValue            = ms[3]
 		value               strings.Builder
 		escaped             bool
 		expectTrailingQuote bool
 	)
 
-	if rawValue[0] == '"' {
+	if strings.HasPrefix(rawValue, "\"") {
 		rawValue = strings.TrimPrefix(rawValue, "\"")
 		expectTrailingQuote = true
 	}

--- a/pkg/labels/parse_test.go
+++ b/pkg/labels/parse_test.go
@@ -191,7 +191,10 @@ func TestMatchers(t *testing.T) {
 		},
 		{
 			input: `job=`,
-			err:   `matcher value is not present: job=`,
+			want: func() []*Matcher {
+				m, _ := NewMatcher(MatchEqual, "job", "")
+				return []*Matcher{m}
+			}(),
 		},
 		{
 			input: `job="value`,

--- a/pkg/labels/parse_test.go
+++ b/pkg/labels/parse_test.go
@@ -190,6 +190,10 @@ func TestMatchers(t *testing.T) {
 			}(),
 		},
 		{
+			input: `job=`,
+			err:   `matcher value is not present: job=`,
+		},
+		{
 			input: `job="value`,
 			err:   `matcher value contains unescaped double quote: "value`,
 		},

